app/controllers/locations_controller.rb

```diff
@@ -41,18 +43,31 @@ def deactivate_confirm
   end
 
   def deactivate
-    @location.run_deactivation_validations!
-
-    if @location.location_deactivation_periods.create!(deactivation_date:) && update_affected_logs
+    if LocationDeactivationPeriod.create!(location_id: @location.id, deactivation_date: params[:deactivation_date]) && update_affected_logs
```

Comment:
```suggestion
    if @location.location_deactivation_periods.create!(deactivation_date: params[:deactivation_date]) && update_affected_logs!
```

Two suggestions here:

1. Come through the association so you don't need to be explicit about the location
2. Make the second method a `!` method to show that it changes state.